src/main/kotlin/week1/racingcar/RacingGame.kt

```diff
@@ -1,36 +1,62 @@
 package week1.racingcar
 
-class RacingGame {
+import kotlin.random.Random
 
-    private var numberOfCars = 0
-    private var timesToRace = 0
+class RacingGame {
 
-    fun start() {
-        InputView().startInput().also {
-            this@RacingGame.apply {
-                numberOfCars = it.numberOfCars
-                timesToRace = it.timesToRace
-            }
-            race()
+    fun getInputResult(): InputResult {
+        return InputView().let {
+            it.getValidateRaceInfo(it.startInput())
         }
     }
 
-    private fun race() {
+    fun startGame(inputResult: InputResult) {
+        val cars = inputResult.cars.map { Car(it) }
+        val timesToRace = inputResult.timesToRace
+        race(cars, timesToRace)
+    }
+
+    private fun race(cars: List<Car>, timesToRace: Int) {
         val resultView = ResultView()
-        val cars = List(numberOfCars) { Car() }
-        (0 until timesToRace).forEach { _ ->
-            cars.forEach {
-                if (isPossibleToDrive()) {
-                    it.go()
-                }
-            }
-            resultView.showResult(cars)
+        repeat(timesToRace) {
+            drive(cars)
+            showResult(resultView, cars)
+        }
+    }
+
+    private fun drive(cars: List<Car>) {
+        cars.forEach {
+            if (isPossibleToDrive()) it.go()
+        }
```

Comment:
자동차가 전진 혹은 정지하는 것은 자동차의 역할인 것 같아요. 

조금 더 객체지향 코드를 작성해보도록 수정해보면 좋을 것 같은데 RacingGame 객체가 직접 전진할 조건을 계산하고 만족하면 자동차 객체를 전진하게 하는 것이 아니라 RacingGame 객체가 자동차 객체에게 `전진 혹은 정지해`라고 요청만하고 그 요청에 대한 수행은 자동차 객체가 스스로 판단해서 수행하도록 변경해볼 수 있을 것 같아요. 

```suggestion
        cars.forEach { it.moveOrNot }
```